src/api/hook.ts

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from "react";
+
+// returns any type T
+// takes in url, options object including request method, headers, body, settings
+const useFetch = <T>(url: string, options: RequestInit = {}) => {
```

Comment:
We can strongly type HTTP methods here, or something of the sort.
```suggestion
type HttpMethod = "GET" | "POST" | "PATCH" | "PUT" | "DELETE";

const useFetch = <T>(url: string, options: RequestInit = {}) => {
```